### PR TITLE
Redirection

### PR DESCRIPTION
**Pull request by TheEgghead27**
(draft PR for easier diff review)

### CODE REVIEW

File: executor.c [hunk 1]
```diff
@@ -59,4 +99,60 @@ void execute(char *command) {
...
+		dup2(backup_stdout, stdout);
+		dup2(backup_stdin, stdin);
 }
+
+// for ref: void run(char *args[16], int input, int output, char *output_file, char *input_file)
+void execute_commands(char *command, char *args[16]) {
+    char *new_args[16];
+    char *input_file = NULL;
+    char *output_file = NULL;
+    int output_mode = 0; // this could be 1 or 2 (see func above to see the diff)
```
Comment (TheEgghead27) [inline executor.c:111]: might be enlightening to refactor as an [enum](https://www.w3schools.com/c/c_enums.php) later

File: executor.c [hunk 2]
```diff
@@ -59,4 +99,60 @@ void execute(char *command) {
...
+    char temp_file[] = "temp.txt";
+    int index_last_command = 0;
+    
+    for (int i = 0; args[i] != NULL; i++) {
+        if (strcmp(args[i], "|") == 0 || strcmp(args[i], "<") == 0 ||
```
Comment (TheEgghead27) [inline executor.c:116]: this is a rather hefty if statement - i feel the code would be clearer and equivalent without it (i am okay with moving `index_last_command = i+1` from the bottom into each case)

File: executor.h [hunk 3]
```diff
@@ -1,4 +1,6 @@
 #ifndef EXECUTOR_H
 #define EXECUTOR_H
 void execute(char *command);
+void execute_commands(char *command, char *args[16]);
```
Comment (TheEgghead27) [inline executor.h:4]: if these functions are not used by main(), there is no need to add them to the headers

i am also mildly displeased at the prospect of having to rejigger all of these arrays...

**Comment by Elias2660** [comment]
pipes work (as of my last check), closes #7

**Comment by Elias2660** [comment]
Tasks: 
- [] finish the piping
- [] remove the logging statements

**Comment by Elias2660** [comment]
> pipes work (as of my last check), closes #7

on the other hand, the cross `cat < sigma.txt | cowsay -f vader  > vader.txt` still needs to be fixed

**Pull request merged**